src/EnergyPlus/Fans.cc

```diff
@@ -111,72 +116,64 @@ void SimulateFanComponents(EnergyPlusData &state,
     // PURPOSE OF THIS SUBROUTINE:
     // This subroutine manages Fan component simulation.
 
-    // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-    int FanNum; // current fan number
-
-    // Obtains and Allocates fan related parameters from input file
-    if (state.dataFans->GetFanInputFlag) { // First time subroutine has been entered
-        GetFanInput(state);
-        state.dataFans->GetFanInputFlag = false;
-    }
-
-    if (CompIndex == 0) {
-        FanNum = Util::FindItemInList(CompName, state.dataFans->Fan, &FanEquipConditions::Name);
-        if (FanNum == 0) {
-            ShowFatalError(state, format("SimulateFanComponents: Fan not found={}", CompName));
-        }
-        CompIndex = FanNum;
-    } else {
-        FanNum = CompIndex;
-        if (FanNum > state.dataFans->NumFans || FanNum < 1) {
-            ShowFatalError(
-                state,
-                format(
-                    "SimulateFanComponents: Invalid CompIndex passed={}, Number of Fans={}, Fan name={}", FanNum, state.dataFans->NumFans, CompName));
-        }
-        if (state.dataFans->CheckEquipName(FanNum)) {
-            if (!CompName.empty() && CompName != state.dataFans->Fan(FanNum).Name) {
-                ShowFatalError(state,
-                               format("SimulateFanComponents: Invalid CompIndex passed={}, Fan name={}, stored Fan Name for that index={}",
-                                      FanNum,
-                                      CompName,
-                                      state.dataFans->Fan(FanNum).Name));
-            }
-            state.dataFans->CheckEquipName(FanNum) = false;
-        }
-    }
-
     // With the correct FanNum Initialize
-    InitFan(state, FanNum, FirstHVACIteration); // Initialize all fan related parameters
+    init(state);
 
+    // if (sizingFlag) return;
```

Comment:
This line is still needed for times when the fan model is called but sizing has not completed.

Reply:
Can confirm with a local Linux build - putting this line back in allows at least the first test failure to pass now.  @amirroth do you want me to push that change and see some new CI results?  Or do you think there's a different (less passive) approach we should employ to avoid calling this function too early?  Let me know if you'd like me to do anything here.

Reply:
Only 3 important ones. 1) add in line 122 above, 2) either add in the `if (CompIndex == 0)` fix at line 3461 for the system fan, or, preferably, initialize `PrimaryAirSystems(AirLoopNum).Branch(BranchNum).Comp(CompNum).CompIndex` in InitAirLoops where the fan index is found so you don't need to check if CompIndex == 0 each iteration, and 3) fix the Custom Check issue with find_byref_bool_override.py since the Fans.cc function and variable names have changed. That should reduce the CI failures to at or near 0.